sys/include/net/gnrc/netdev2.h

```diff
@@ -102,6 +103,16 @@ typedef struct gnrc_netdev2 {
      * @brief general information for the MAC protocol
      */
     uint16_t mac_info;
+
+    /**
+     * @brief device's l2 address
+     */
+    uint8_t  l2_addr[IEEE802154_LONG_ADDRESS_LEN];
```

Comment:
@zhuoshuguo because an IEEE 802.15.4 device driver is supposed to call `netdev2_ieee802154_set` in its `set` method to reduce code duplication. What you propose would be a loop ;-).